src/components/SummaryNameMapModal.tsx

```diff
@@ -0,0 +1,118 @@
+import {
+    Select,
+    Button,
+    ModalContent,
+    ModalHeader,
+    ModalCloseButton,
+    ModalBody,
+    ModalFooter,
+    FormControl,
+    FormLabel,
+    ModalOverlay,
+    VStack
+} from '@chakra-ui/react';
+import React, { useEffect, useState } from 'react';
+import trpc from "../trpc";
+import ModalWithBackdrop from 'components/ModalWithBackdrop';
+import ReactMarkdown from 'react-markdown';
+import { MinUser } from 'shared/User';
+import { SummaryNameMap } from 'shared/Summary';
+
+export function SummaryNameMapModal({ nameMap, groupUsers, onClose }: {
+    nameMap: SummaryNameMap | undefined
+    groupUsers: MinUser[]
+    onClose: () => void,
+}) {
+    // UserMap = { [handlebar] : { name: [userName], id: [userId] }}
```

Comment:
okay I think you will need to change the `getNameMap` to return `SummaryNameMap` object which contains userids and not user names. This will greatly simplify your code and makes more sense.

Reply:
also change the parameter type of `updateNameMap` to `SummaryNameMap`. All these backend changes should be done in a separate PR.